File: PSET1/mario-less/mario.c
#include <stdio.h>
#include <cs50.h>

//Prototype
void draw(int height, int initial_height);

int main(void)
{
    int h;
    //Prompt user for height, number should be 1-8
    do
    {
        h = get_int("Enter pyramid height(1-8): \n");
    }
    while (h < 1 || h > 8);

    draw(h, h);
}

void draw(int height, int initial_height)
{
    if (height == 0)
    {
        return;
    }

    //Drawing the pyramid
    draw(height - 1, initial_height);

    for (int i = 0; i < initial_height - height; i++)
    {
        printf(" ");
    }
    for (int i = 0; i < height; i++)
    {
        //Adding the hashes
        printf("#");
    }
    printf("\n");
}

File: PSET6/mario-less/mario.py
# prototype
def main():

    height = get_height()
    draw(height, height)


# printing the pyramid
def draw(height, h):
    if height == 0:
        return
    draw(height - 1, h)
    print(" " * (h - height), end='')
    print("#" * height)


# prompting the user for height
def get_height():
    while True:
        try:
            height = int(input("Height: "))
            if (height > 0 and height < 9):
                break
        except ValueError:
            None

    return height


# retun main function
main()
